buildshiprun/handler.go

```diff
@@ -42,6 +42,7 @@ func Handle(req []byte) string {
 	log.Printf("%d env-vars for %s", len(event.Environment), serviceValue)
 
 	status := sdk.BuildStatus(event, sdk.EmptyAuthToken)
+	status.SetHmacKey(os.Getenv("github_webhook_secret"))
```

Comment:
I would have expected to see the key not set like this, but just passed in the final step where it "sends it" or "submits it".

Reply:
i.e. your helper method named reportStatus

```
reportStatus(status, key)
```

Reply:
`status.Report(gateway, key)` is the new way. The helper method `reportStatus()` reads the key and forward it.